cockatrice/src/window_main.cpp

```diff
@@ -319,6 +319,9 @@ void MainWindow::loginError(Response::ResponseCode r, QString reasonStr, quint32
         case Response::RespClientIdRequired:
             QMessageBox::critical(this, tr("Error"), tr("This server requires client ID's. Your client is either failing to generate an ID or you are running a modified client.\nPlease close and reopen your client to try again."));
             break;
+        case Response::RespContextError:
+            QMessageBox::critical(this, tr("Error"), tr("An internal error has occurred, please try closing and reopening your client and try again. If the error persists try updating your client to the most recent build and if need be contact your software provider."));
```

Comment:
I must not have reviewed the change enough. You can add new fields to
existing messages without breaking them

On Sun, Aug 9, 2015, 11:53 PM woogerboy21 <notifications@github.com> wrote:

> In cockatrice/src/window_main.cpp
> <https://github.com/Cockatrice/Cockatrice/pull/1350#discussion_r36600508>:
>
> > @@ -319,6 +319,9 @@ void MainWindow::loginError(Response::ResponseCode r, QString reasonStr, quint32
> >          case Response::RespClientIdRequired:
> >              QMessageBox::critical(this, tr("Error"), tr("This server requires client ID's. Your client is either failing to generate an ID or you are running a modified client.\nPlease close and reopen your client to try again."));
> >              break;
> > +        case Response::RespContextError:
> > +            QMessageBox::critical(this, tr("Error"), tr("An internal error has occurred, please try closing and reopening your client and try again. If the error persists try updating your client to the most recent build and if need be contact your software provider."));
>
> Your request to have the clientid check in the login container (#1344
> <https://github.com/Cockatrice/Cockatrice/pull/1344>) forces a client
> update [image: :confused:]
> When I rolled the change into the server I took the check out to allow
> older clients connectivity until the most recent release was published and
> at a later date when I upgrade the server again I'll keep it in.
>
> —
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/Cockatrice/Cockatrice/pull/1350/files#r36600508>.
>